File: ios/BelleUtility.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BelleUtility, NSObject)

RCT_EXTERN_METHOD(scheduleReminders:(NSString*)groupId title: (NSString*)title content: (NSString*)content firedDateInMs:(NSNumber*)firedDate repeatType: (NSString*)repeatType repeatStep: (NSNumber *)repeatStep extraData: (NSString*)extraData)
RCT_EXTERN_METHOD(cancelReminders:(NSString*)groupId)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/BelleUtility.swift
import Foundation

@objc(BelleUtility)
class BelleUtility: NSObject {

    @available(iOS 10.0, *)
    @objc(cancelReminders:)
    func cancelReminders(groupId: String?) -> Void {
        LocalNotificationHelper.cancelReminders(groupId: groupId)
    }
    
    @available(iOS 10.0, *)
    @objc(scheduleReminders:title:content:firedDateInMs:repeatType:repeatStep:extraData:)
    func scheduleReminders(groupId: String, title: String, content: String, firedDateInMs: Double, repeatType: String, repeatStep: Double, extraData: String?) -> Void {
        LocalNotificationHelper.scheduleReminders(groupId: groupId, title: title, content: content, firedDateInMs: firedDateInMs, repeatType: repeatType, repeatStep: repeatStep, extraData: extraData)
    }
}
